kmir/src/kmir/convert_json/convert.py

```diff
@@ -22,31 +20,57 @@ def _unimplemented() -> NoReturn:
 
 def maybe_byte_from_dict(js: int | None) -> KApply:
     if js is None:
-        return KApply('noByte_BODY_MaybeByte', ())
+        return KApply('noByte', ())
 
-    return KApply('someByte(_)_TYPES_MaybeByte_Int', (KToken(str(js), KSort('Int'))))
+    return KApply('someByte', (KToken(str(js), KSort('Int'))))
 
 
 def bytes_from_dict(js: Sequence[int]) -> KApply:
     if len(js) == 0:
-        return KApply('.List{"___TYPES_Bytes_MaybeByte_Bytes"}_Bytes', ())
+        return KApply('.maybeBytes', ())
+
+    return KApply('maybeBytes', (maybe_byte_from_dict(js[0]), bytes_from_dict(js[1:])))
 
-    return KApply('___TYPES_Bytes_MaybeByte_Bytes', (maybe_byte_from_dict(js[0]), bytes_from_dict(js[1:])))
+
+def provenance_map_entry_from_dict(js: Sequence[object]) -> KApply:
+    match js:
+        case [int(size), {'Memory': dict()}]:
+            # TODO: we actually want to parse a provenance map here; need to
```

Comment:
As you noticed, there is a discrepancy here, between the expected AST and the serialized output. My guess is that this may be due to the `allocId` being materialized, while the AST representation expects it to be interned. I am not sure what needs to be updated here (the output of the compiler or the AST). but we will make sure to address this.